internal/dag/gatewayapi_processor.go

```diff
@@ -1119,9 +1119,47 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1beta1.HTTPRou
 				// Get the redirect filter if there is one. Note that per Gateway API
 				// docs, "specifying a core filter multiple times has unspecified or
 				// custom conformance.", here we choose to just select the first one.
+				if pathRewritePolicy != nil {
+					continue
+				}
+
 				if redirect == nil && filter.RequestRedirect != nil {
 					redirect = filter.RequestRedirect
 				}
+
+				if redirect.Path == nil {
+					continue
+				}
+
+				var prefixRewrite, fullPathRewrite string
+
+				switch redirect.Path.Type {
+				case gatewayapi_v1beta1.PrefixMatchHTTPPathModifier:
+					if filter.RequestRedirect.Path.ReplacePrefixMatch == nil || len(*filter.RequestRedirect.Path.ReplacePrefixMatch) == 0 {
+						prefixRewrite = "/"
+					} else {
+						prefixRewrite = *filter.RequestRedirect.Path.ReplacePrefixMatch
+					}
+				case gatewayapi_v1beta1.FullPathHTTPPathModifier:
+					if filter.RequestRedirect.Path.ReplaceFullPath == nil || len(*filter.RequestRedirect.Path.ReplaceFullPath) == 0 {
+						fullPathRewrite = "/"
+					} else {
+						fullPathRewrite = *filter.RequestRedirect.Path.ReplaceFullPath
+					}
+				default:
+					routeAccessor.AddCondition(
+						gatewayapi_v1beta1.RouteConditionAccepted,
+						metav1.ConditionFalse,
+						gatewayapi_v1beta1.RouteReasonUnsupportedValue,
+						fmt.Sprintf("HTTPRoute.Spec.Rules.Filters.RequestRedirect.Path.Type: invalid type %q: only ReplacePrefixMatch and ReplaceFullPath are supported.", filter.RequestRedirect.Path.Type),
+					)
+					continue
+				}
+
+				if redirect != nil {
+					redirect.Path.ReplaceFullPath = &fullPathRewrite
+					redirect.Path.ReplacePrefixMatch = &prefixRewrite
+				}
```

Comment:
@skriss should this not be enough as if `redirect != nil` we are calling this some lines below
https://github.com/projectcontour/contour/blob/e511e8129006b371a81f8e9df5f982bb0b67baab/internal/dag/gatewayapi_processor.go#L1230-L1254

Reply:
> by the way, apologies for being so late on this, I was busy in college work and with some other organisations as i was trying to get into LFX and GSoC for internships opportunities

Not a problem, hope all turns out well for you!